app/controllers/users/registrations_controller.rb

```diff
@@ -1,45 +1,36 @@
 # frozen_string_literal: true
 
 class Users::RegistrationsController < Devise::RegistrationsController
-  def start_registration; end
+  def info; end
 
-  def confirm_school
-    @schools = School.where(id: params["school_ids"])
-    @email = params["email"]
+  def new
+    @user = User.new
   end
 
-  def check_email
+  def check_details
+    @urn = params["induction_coordinator_profile"]["school_urn"]
     @email = params["induction_coordinator_profile"]["email"]
-    @user = User.find_by(email: @email)
-
-    if @user
-      flash[:notice] = "You already have an account. Sign in"
-      redirect_to controller: "/users/sessions", action: :new, email: @email
-    else
-      @schools = School.where("'#{email_domain}' = ANY (domains)")
+    @user = User.find_or_initialize_by(email: @email)
+    @school = School.where(urn: @urn).where("'#{email_domain}' = ANY (domains)").first
 
-      if @schools.any?
-        handle_matching_schools
-      else
-        redirect_to users_check_email_path, alert: "No schools matched your email"
-      end
-    end
-  end
+    render :user_already_registered and return if @user.persisted?
+    render_start_registration and return if @school.nil?
+    render :school_not_eligible and return unless @school.eligible?
+    render :school_not_registered and return if @school.not_registered?
+    render :school_fully_registered and return if @school.fully_registered?
 
-  def new
-    @email = params[:email]
-    @school = School.find(params[:school_id])
-    super
+    render :school_partially_registered if @school.partially_registered?
   end
 
   def create
     validate_creation_parameters
+    build_resource(sign_up_params)
```

Comment:
What does `build_resource` do?

Reply:
It's like User.build(params) but it comes from devise. I had to override devise on this controller action so that I can render a template rather than a flash message.